libs/base/Data/Bits.idr

```diff
@@ -23,6 +24,13 @@ fromNat :  (k : Nat)
         -> Subset Nat (`LT` n)
 fromNat k = Element k (ltReflectsLT k n prf)
 
+||| Utility for implementing `bitsToIndex` in `FiniteBits`.
+export
+fromFin : Fin n -> Subset Nat (`LT` n)
+fromFin FZ     = Element 0 (LTESucc LTEZero)
```

Comment:
I believe the right place for this function is in `Data.Fin`, somewhere near for `finToNat`, named like `finToNat'` or `finToBoundedNat`.